### DISCUSSION TITLE
Occasionally, the web page cannot load the list, displaying:There are no items to list.

### DISCUSSION THREAD

**Discussion by lpcy** [dani-garcia/vaultwarden#3448] [Q&A]
### Subject of the issue
I added an item. But every time the page is refreshed and a password is entered, the page may display this item normally or empty.

### Deployment environment
docker image: vaultwarden/server:1.28.1
env: WEBSOCKET_ENABLED: true
Clients used:  web vault
Reverse proxy: Nginx / https

### Troubleshooting data
I tried to solve this problem and found that it may be caused by asynchronous loading of the page.

**Comment by BlackDex** [comment]
We need a bit more info.
Do you have any logs of both the reverse proxy and vaultwarden during the time this happens.
Are there any Javascript errors/warning in the F12 Developer Console?

This is too vague to do anything with.

**Comment by lpcy** [comment]
I didn't see any errors on the console, nginx, and server. It's very easy to reproduce on my old computer, but new computer rarely appear. I tried to figure out where the problem was, but unfortunately, I am not familiar with the front-end Angular framework. I am only aware that this is due to an asynchronous issue:
1. After entering the password to unlock, the doUnlockWithMasterPassword method is called, and after that, I realized that two lines were starting to execute:
   1) vault.component.ts : ```ngOnInit>getEmailVerified>decodeToken> ...not finished```
   2) vault-items.commpnent.ts: ```set activeFilter > reload > load > applyFilter >search > doSearch > getAllDecrypted```
   I'm not sure why both lines are executing at the same time.
2. After the line 2 completes **getAllDecrypted**, the line 1 will continue to execute ```fullSync>setEncryptedCiphers```, after that, ```data.ciphers.encrypted``` only has data. So the first time line 2 **getAllDecrypted** was not able to obtain complete data.
3. When calling the **getAllDecrypted** method for the second time, if the **getAllDecrypted** cache in sequence.ts exists (i.e. **getAllDecrypted** has not completed execution), the cache will be taken out and empty. If the **getAllDecrypted** cache has expired (the previous **getAllDecrypted** execution completed), the **getAllDecrypted** will be re executed to obtain the value. It seems that **getAllDecrypted** is the key to whether the list can be displayed. These two situations result in randomness due to asynchronous issues.

**Comment by BlackDex** [comment]
I'm afraid that this project can't help you. You mentioned an old computer. Now that is very subjective i think. But decrypting there vault data takes some cpu power.

If Vaultwarden returns all the data without issues the problem is on the client side, which is not what we develop or maintain.

One thing you maybe can try is to lower your kdf iterations. You can do this via settings. I would suggest to do that on the newer computer.

**Reply by lpcy** [reply]
Thank you for your reply. I mainly want to describe that the reproduction of the problem may be related to computing power, and my old computer also has the Intel 8th generation i7 core. Because I have encountered this situation too many times, I estimate many others have already encountered it, but I have not found any similar discussions.
